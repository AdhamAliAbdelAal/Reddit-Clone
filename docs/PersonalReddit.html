<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="utf-8">
    <title>JSDoc: Class: PersonalReddit</title>

    <script src="scripts/prettify/prettify.js"> </script>
    <script src="scripts/prettify/lang-css.js"> </script>
    <!--[if lt IE 9]>
      <script src="//html5shiv.googlecode.com/svn/trunk/html5.js"></script>
    <![endif]-->
    <link type="text/css" rel="stylesheet" href="styles/prettify-tomorrow.css">
    <link type="text/css" rel="stylesheet" href="styles/jsdoc-default.css">
</head>

<body>

<div id="main">

    <h1 class="page-title">Class: PersonalReddit</h1>

    




<section>

<header>
    
        <h2><span class="attribs"><span class="type-signature"></span></span>PersonalReddit<span class="signature">()</span><span class="type-signature"> &rarr; {React.Component}</span></h2>
        
    
</header>

<article>
    <div class="container-overview">
    
        

    

    
    <h4 class="name" id="PersonalReddit"><span class="type-signature"></span>new PersonalReddit<span class="signature">()</span><span class="type-signature"> &rarr; {React.Component}</span></h4>
    

    



<div class="description">
    <p>This component contains info about personal
Reddit frontpage.</p>
</div>













<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="components_HomePage_HomePageContainer_PersonalReddit_PersonalReddit.jsx.html">components/HomePage/HomePageContainer/PersonalReddit/PersonalReddit.jsx</a>, <a href="components_HomePage_HomePageContainer_PersonalReddit_PersonalReddit.jsx.html#line15">line 15</a>
    </li></ul></dd>
    

    

    

    
</dl>















<h5>Returns:</h5>

        
<div class="param-desc">
    <p>Buttons to show info about communities</p>
</div>



<dl>
    <dt>
        Type
    </dt>
    <dd>
        
<span class="param-type">React.Component</span>


    </dd>
</dl>

    





    
    </div>

    

    

    

    

    

    

    

    

    

    
</article>

</section>




</div>

<nav>
    <h2><a href="index.html">Home</a></h2><h3>Classes</h3><ul><li><a href="BackToTop.html">BackToTop</a></li><li><a href="global.html#Communities">Communities</a></li><li><a href="CommunityItem.html">CommunityItem</a></li><li><a href="CommunityList.html">CommunityList</a></li><li><a href="CreatePostContainer.html">CreatePostContainer</a></li><li><a href="CreatePostForm.html">CreatePostForm</a></li><li><a href="CreatePostInHome.html">CreatePostInHome</a></li><li><a href="FirstParty.html">FirstParty</a></li><li><a href="HomePageContainer.html">HomePageContainer</a></li><li><a href="HomePageFooter.html">HomePageFooter</a></li><li><a href="JoinButton.html">JoinButton</a></li><li><a href="LogInPage.html">LogInPage</a></li><li><a href="global.html#MainContent">MainContent</a></li><li><a href="NotificationCategories.html">NotificationCategories</a></li><li><a href="NotificationImages.html">NotificationImages</a></li><li><a href="NotificationsBody.html">NotificationsBody</a></li><li><a href="NotificationsHeader.html">NotificationsHeader</a></li><li><a href="PersonalReddit.html">PersonalReddit</a></li><li><a href="global.html#Post">Post</a></li><li><a href="PostHeader.html">PostHeader</a></li><li><a href="PostList.html">PostList</a></li><li><a href="PostReactions.html">PostReactions</a></li><li><a href="PostSubmission.html">PostSubmission</a></li><li><a href="PostTags.html">PostTags</a></li><li><a href="PostTypes.html">PostTypes</a></li><li><a href="PostsClassification.html">PostsClassification</a></li><li><a href="Reactions.html">Reactions</a></li><li><a href="RedditPremium.html">RedditPremium</a></li><li><a href="global.html#SideBar">SideBar</a></li><li><a href="SubredditsMenu.html">SubredditsMenu</a></li></ul><h3>Global</h3><ul><li><a href="global.html#About">About</a></li><li><a href="global.html#AddList">AddList</a></li><li><a href="global.html#AddSector">AddSector</a></li><li><a href="global.html#BackHomeBottun">BackHomeBottun</a></li><li><a href="global.html#Comments">Comments</a></li><li><a href="global.html#CommentsHeader">CommentsHeader</a></li><li><a href="global.html#Content">Content</a></li><li><a href="global.html#EmptyContent">EmptyContent</a></li><li><a href="global.html#Filter">Filter</a></li><li><a href="global.html#Flirt">Flirt</a></li><li><a href="global.html#FormDialog">FormDialog</a></li><li><a href="global.html#Header">Header</a></li><li><a href="global.html#LoadingPage">LoadingPage</a></li><li><a href="global.html#LogIn">LogIn</a></li><li><a href="global.html#Moderation">Moderation</a></li><li><a href="global.html#Moderators">Moderators</a></li><li><a href="global.html#Notifications">Notifications</a></li><li><a href="global.html#PostFooter">PostFooter</a></li><li><a href="global.html#PostFooterList">PostFooterList</a></li><li><a href="global.html#PostSide">PostSide</a></li><li><a href="global.html#Posts">Posts</a></li><li><a href="global.html#PostsFilteredTap">PostsFilteredTap</a></li><li><a href="global.html#PostsTap">PostsTap</a></li><li><a href="global.html#Sidebar">Sidebar</a></li><li><a href="global.html#SignUp">SignUp</a></li><li><a href="global.html#SignUpEmail">SignUpEmail</a></li><li><a href="global.html#SignUpUsername">SignUpUsername</a></li><li><a href="global.html#ThirdParty">ThirdParty</a></li><li><a href="global.html#ThirdPartyButton">ThirdPartyButton</a></li><li><a href="global.html#ThirdPartyContainer">ThirdPartyContainer</a></li><li><a href="global.html#UserInfo">UserInfo</a></li><li><a href="global.html#calculateTime">calculateTime</a></li><li><a href="global.html#checkPassword">checkPassword</a></li><li><a href="global.html#checkUserName">checkUserName</a></li><li><a href="global.html#refreshUsernames">refreshUsernames</a></li><li><a href="global.html#responseFacebook">responseFacebook</a></li><li><a href="global.html#responseGoogleSuccess">responseGoogleSuccess</a></li><li><a href="global.html#signUp">signUp</a></li></ul>
</nav>

<br class="clear">

<footer>
    Documentation generated by <a href="https://github.com/jsdoc/jsdoc">JSDoc 4.0.0</a> on Thu Nov 10 2022 20:06:19 GMT+0200 (Eastern European Standard Time)
</footer>

<script> prettyPrint(); </script>
<script src="scripts/linenumber.js"> </script>
</body>
</html>