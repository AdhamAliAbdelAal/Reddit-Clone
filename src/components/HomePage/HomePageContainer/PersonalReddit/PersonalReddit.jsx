import { Divider, Typography } from '@mui/material';
import {
  RedditPersonal, Paragraph, RedditImage, MiddleBox, UpperImage, CustomLink,
} from './styles';
import RedditButton from '../../../RedditButton/RedditButton';
import FormDialog from './PopUpSubReddit/PopUp';
/**
 * This component contains info about personal
 * Reddit frontpage.
 *
 * @component PersonalReddit
 * @property {string} title -Title of personal reddit
 * @property {string} paragraph -Paragraph of personal reddit
 * @property {string} image -Image of personal reddit
 * @returns {React.Component}  Buttons to show info about communities
 */

function PersonalReddit(props) {
  const { title, paragraph, image } = props;
  return (
    <RedditPersonal>
      <UpperImage src={image} />
      <MiddleBox>
        <RedditImage src="https://www.redditstatic.com/desktop2x/img/id-cards/snoo-home@2x.png" />
        <Typography
          component="p"
          sx={{
            fontWeight: 500,
            marginLeft: 7,
            pb: 0.8,
            pt: 3,
          }}
        >
          {title}
        </Typography>
      </MiddleBox>
      <Paragraph>
        {paragraph}
      </Paragraph>
      <Divider sx={{ mt: 1.5 }} />
      <CustomLink to="/submit">
        <RedditButton variant="contained" padding="4px" fontSize={15} fontWeight="bold" sx={{ mt: 1.8 }}>
          create post
        </RedditButton>
      </CustomLink>
      <FormDialog />
    </RedditPersonal>
  );
}

export default PersonalReddit;
