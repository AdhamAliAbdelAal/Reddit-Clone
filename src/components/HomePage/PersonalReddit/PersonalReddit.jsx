import { Divider, Typography } from '@mui/material';
import {
  RedditPersonal, Paragraph, RedditImage, MiddleBox, UpperImage, CustomLink,
} from './styles';
import RedditButton from '../../RedditButton/RedditButton';

function PersonalReddit() {
  return (
    <RedditPersonal>
      <UpperImage src="https://www.redditstatic.com/desktop2x/img/id-cards/home-banner@2x.png" />
      <MiddleBox>
        <RedditImage src="https://www.redditstatic.com/desktop2x/img/id-cards/snoo-home@2x.png" />
        <Typography
          component="p"
          sx={{
            fontWeight: 500,
            textTransform: 'capitalize',
            marginLeft: 7,
            pb: 0.8,
            pt: 3,
          }}
        >
          home
        </Typography>
      </MiddleBox>
      <Paragraph>
        Your personal Reddit frontpage. Come here to check in with your favorite communities.
      </Paragraph>
      <Divider sx={{ mt: 1.5 }} />
      <CustomLink to="/submit">
        <RedditButton variant="contained" padding="4px" fontSize={15} fontWeight="bold" sx={{ mt: 1.8 }}>
          create post
        </RedditButton>
      </CustomLink>
      <CustomLink>
        <RedditButton variant="outlined" padding="4px" fontSize={15} fontWeight="bold" sx={{ mt: 1.8 }}>
          create community
        </RedditButton>
      </CustomLink>
    </RedditPersonal>
  );
}

export default PersonalReddit;
