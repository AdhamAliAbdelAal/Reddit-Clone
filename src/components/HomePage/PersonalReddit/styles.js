import { Box } from '@mui/material';
import { styled } from '@mui/system';

export const RedditPersonal = styled('div')(() => ({
  padding: 12,
  display: 'flex',
  flexDirection: 'column',
  backgroundColor: '#fff',
  marginTop: 16,
  borderRadius: 7,
}));

export const Paragraph = styled(Box)(() => ({
  fontSize: 14,
  fontWeight: '400',
  lineHeight: '21px',
  marginTop: 12,
}));

export const MiddleBox = styled(Box)(() => ({
  position: 'relative',
}));
export const RedditImage = styled('img')(() => ({
  width: 40,
  height: 68,
  position: 'absolute',
  bottom: 0,
}));
export const UpperImage = styled('img')(() => ({
  margin: '-12px -12px 0 -12px',
}));
