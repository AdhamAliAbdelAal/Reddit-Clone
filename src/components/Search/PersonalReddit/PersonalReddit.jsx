import { Divider } from '@mui/material';
import {
  RedditPersonal, Paragraph, RedditImage, MiddleBox, UpperImage, CustomLink, Home,
} from './styles';
import RedditButton from '../RedditButton/RedditButton';

function PersonalReddit() {
  return (
    <RedditPersonal>
      <UpperImage src="https://www.redditstatic.com/desktop2x/img/id-cards/home-banner@2x.png" />
      <MiddleBox>
        <RedditImage src="https://www.redditstatic.com/desktop2x/img/id-cards/snoo-home@2x.png" />
        <Home
          component="p"
        >
          home
        </Home>
      </MiddleBox>
      <Paragraph>
        Your personal Reddit frontpage. Come here to check in with your favorite communities.
      </Paragraph>
      <Divider sx={{ mt: 1.5 }} />
      <CustomLink>
        <RedditButton variant="outlined" padding="4px" fontSize={15} fontWeight="bold" sx={{ mt: 1.8 }}>
          create community
        </RedditButton>
      </CustomLink>
    </RedditPersonal>
  );
}

export default PersonalReddit;
