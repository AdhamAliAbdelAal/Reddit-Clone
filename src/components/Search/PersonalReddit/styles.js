import { Box, Typography } from '@mui/material';
import { styled } from '@mui/material/styles';
import { Link } from 'react-router-dom';

export const RedditPersonal = styled('div')(() => ({
  padding: 12,
  display: 'flex',
  flexDirection: 'column',
  backgroundColor: '#fff',
  marginTop: 16,
  borderRadius: 7,
  border: '1px solid #ccc',
}));

export const Paragraph = styled(Box)(() => ({
  fontSize: 14,
  fontWeight: '400',
  lineHeight: '21px',
  marginTop: 70,
}));

export const MiddleBox = styled(Box)(() => ({
  position: 'relative',
}));
export const RedditImage = styled('img')(() => ({
  width: 70,
  height: 126,
  position: 'absolute',
  bottom: -67,
  transform: 'translateX(-50%)',
  left: '50%',

}));
export const UpperImage = styled('img')(() => ({
  margin: '-12px -12px 0 -12px',
}));
export const CustomLink = styled(Link)(() => ({
  textDecoration: 'none',
  display: 'flex',
  flexDirection: 'column',
}));
export const Home = styled(Typography)(() => ({
  fontWeight: 500,
  textTransform: 'capitalize',
  marginLeft: 56,
  paddingBottom: 6.4,
  paddingTop: 24,
}));
