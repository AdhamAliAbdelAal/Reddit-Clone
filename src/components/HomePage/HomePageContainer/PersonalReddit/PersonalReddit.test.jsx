import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';
import { MemoryRouter as Router } from 'react-router-dom';
import PersonalReddit from './PersonalReddit';

describe(PersonalReddit, () => {
  it('PersonalReddit renders correctly', () => {
    const tree = renderer
      .create(
        <Router>
          <PersonalReddit />
          ,
        </Router>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
