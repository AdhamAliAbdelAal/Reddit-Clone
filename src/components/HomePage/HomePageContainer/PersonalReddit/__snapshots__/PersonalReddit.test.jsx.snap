// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PersonalReddit PersonalReddit renders correctly 1`] = `
Array [
  <div
    className="css-170bi6"
  >
    <img
      className="css-1fb17k2"
      src="https://www.redditstatic.com/desktop2x/img/id-cards/home-banner@2x.png"
    />
    <div
      className="MuiBox-root css-19sk4h4"
    >
      <img
        className="css-1bg7efm"
        src="https://www.redditstatic.com/desktop2x/img/id-cards/snoo-home@2x.png"
      />
      <p
        className="MuiTypography-root MuiTypography-body1 css-gkoyxa-MuiTypography-root"
      >
        home
      </p>
    </div>
    <div
      className="MuiBox-root css-zlg1p4"
    >
      Your personal Reddit frontpage. Come here to check in with your favorite communities.
    </div>
    <hr
      className="MuiDivider-root MuiDivider-fullWidth css-v9rab2-MuiDivider-root"
    />
    <a
      className="css-jta5yf"
      href="/submit"
      onClick={[Function]}
    >
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-1b07p3c-MuiButtonBase-root-MuiButton-root"
        disabled={false}
        fontSize={15}
        fontWeight="bold"
        onBlur={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        padding="4px"
        tabIndex={0}
        type="button"
      >
        create post
      </button>
    </a>
    <a
      className="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-jf1sgt-MuiTypography-root-MuiLink-root"
      onBlur={[Function]}
      onFocus={[Function]}
    >
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium css-1gr231h-MuiButtonBase-root-MuiButton-root"
        data-testid="btn"
        disabled={false}
        fontSize={15}
        fontWeight="bold"
        id="popup-form-button"
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        padding="4px"
        tabIndex={0}
        type="button"
      >
        create community
      </button>
    </a>
    <div
      className="MuiBox-root css-cxf04i"
    />
  </div>,
  ",",
]
`;
